skelebot/components/artifactory.py

```diff
@@ -32,18 +33,43 @@ def pushArtifact(artifactFile, user, token, file, url, force):
         os.remove(file)
         raise
 
-def pullArtifact(user, token, file, url):
+def pullArtifact(user, token, file, url, override, original):
     """Pulls the given file from the url with the provided user/token auth"""
 
     if (artifactory.ArtifactoryPath(url, auth=(user, token)).exists()):
         print("Pulling {file} from {url}".format(file=file, url=url))
         path = artifactory.ArtifactoryPath(url, auth=(user, token))
         with path.open() as fd:
-            with open(file, "wb") as out:
+            dest = original if (override) else file
```

Comment:
What happens if the user specifies `override` but there's no artifact in the config file, or the artifact path doesn't exist yet? Can that even happen, i.e., can `original` be None?

Reply:
Also, there must be an artifact in the config file. They won't be able to even issue this command without it.

Reply:
Ah, so `original` will never be None. That makes sense.